runtime/tools/python/ttrt/common/memory.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import json
+import importlib.machinery
+import sys
+import signal
+import os
+import io
+import subprocess
+import time
+import socket
+from pkg_resources import get_distribution
+import shutil
+import atexit
+import re
+from functools import partial
+
+from ttrt.common.util import *
+
+
+class MemoryRuntimeConfig:
+    def __init__(self, device=None, artifact_dir="", tt_metal_home_path=""):
+        self.device = device
+        self.artifact_dir = artifact_dir
+        self.tt_metal_home_path = tt_metal_home_path
+
+
+def memory(memory_runtime_config, binary, program_context, op_context):
+    device = memory_runtime_config.device
+
+    print("-----------executing memory dump-----------")
+
+    try:
+        import ttrt.runtime
+        import ttrt.binary
+
+        device.dump_memory_report()
+
+        op_debug_str = ttrt.runtime.get_op_debug_str(op_context)
+
+        # find matching golden tensor based on loc in op debug string
+        match = re.search(r"loc\(([^)]+)\)", op_debug_str)
+
+        if not match:
+            print(f"debug_str={op_debug_str}")
+            print("No location found in debug string - skipping memory dump")
```

Comment:
Maybe the prints should be logger statements? It's unclear that we want to always spew this information, I could imagine it being very noisy for big models with lots of ops

Reply:
agreed. I'll keep relevant print statements in logging.debug